service/build.gradle

```diff
@@ -34,6 +34,7 @@ dependencies {
     implementation 'bio.terra:terra-common-lib'
     implementation 'org.apache.commons:commons-dbcp2:2.9.0'
     implementation 'org.springframework.boot:spring-boot-starter-data-jdbc:2.7.0'
+    implementation 'org.springframework.boot:spring-boot-starter-data-jpa:2.7.0'
```

Comment:
This dependency block could be improved.

- most dependencies here (like this one) don't need a version tag, because they're configured by the spring dependency plugin
- some dependencies aren't used. Maybe someday they'll be needed; I would remove them until they are.

This block works when I run `service:bootRun` and `check` locally on this branch:
```
dependencies {

    implementation 'bio.terra:terra-common-lib'
    implementation 'org.apache.commons:commons-dbcp2'
    implementation 'org.springframework.boot:spring-boot-starter-data-jdbc'
    implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
    implementation 'org.springframework.boot:spring-boot-starter-web'
    implementation 'org.springframework.retry:spring-retry'

    implementation 'org.broadinstitute.dsde.workbench:sam-client_2.12:0.1-61135c7'
    implementation 'javax.ws.rs:javax.ws.rs-api:2.1.1'
    implementation 'org.postgresql:postgresql'
    implementation 'org.springframework.boot:spring-boot-starter-validation'
    implementation 'org.springframework.boot:spring-boot-starter-thymeleaf'

    // https://projectlombok.org
    compileOnly 'org.projectlombok:lombok:1.18.20'
    annotationProcessor 'org.projectlombok:lombok:1.18.20'

    liquibaseRuntime 'org.liquibase:liquibase-core'
    liquibaseRuntime 'info.picocli:picocli:4.6.1'
    liquibaseRuntime 'ch.qos.logback:logback-classic'

    annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"

    // Spotbugs dependencies
    compileOnly "com.github.spotbugs:spotbugs-annotations:${spotbugs.toolVersion.get()}"
    spotbugs "com.github.spotbugs:spotbugs:${spotbugs.toolVersion.get()}"

    testImplementation 'org.junit.jupiter:junit-jupiter-api'
    testImplementation('org.springframework.boot:spring-boot-starter-test') {
        // Fixes warning about multiple occurrences of JSONObject on the classpath
        exclude group: 'com.vaadin.external.google', module: 'android-json'
    }
    testImplementation 'org.mockito:mockito-inline'
}
```

Reply:
where is the spring dependency plugin being pulled in?  This whole gradle on top of spring dependency magic throws me for a loop.

Is there an easy way to check if a dependency is being used or do i just delete a line and run stuff and see if it breaks?